### PR TITLE
moved labels to the same device as logits for BLOOM, GPT Neo, GPT NeoX, RoBERTa and VIT models

### PR DESCRIPTION
**Pull request by iamarunbrahma**
# What does this PR do?

As suggested in the https://github.com/huggingface/transformers/issues/22561, moved labels to the same device as logits for `BLOOM`, `GPT Neo`, `GPT NeoX`, `RoBERTa` and `VIT` models.

@sgugger Could you review this once?

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Nice, thanks a lot!
